### ISSUE TITLE
No null safety for tappable cells

### ISSUE THREAD

**Issue by NatashaFatima** [ZainUlMustafa/Fluttable#5]
Tappable cells throw an exception when tapped if the onCellTap callback is not provided. 

Logs:-
```
The following NoSuchMethodError was thrown while handling a
gesture:
                    The method 'call' was called on null.
                    Receiver: null
                    Tried calling: call()
                    
                    When the exception was thrown, this was the stack:
                    #0      _FluttableState.detectOnCellTap
                    (package:qurbaniapp/Cidgets/fluttable.dart:475:21)
                    #1      _FluttableState.relevantCellDetector.<anonymous
closure>
                    (package:qurbaniapp/Cidgets/fluttable.dart:442:17)
                    #2      _InkResponseState._handleTap
                    (package:flutter/src/material/ink_well.dart:991:20)
                    #3      GestureRecognizer.invokeCallback
                    (package:flutter/src/gestures/recognizer.dart:182:24)
                    #4      TapGestureRecognizer.handleTapUp
                    (package:flutter/src/gestures/tap.dart:607:11)
                    #5      BaseTapGestureRecognizer._checkUp
                    (package:flutter/src/gestures/tap.dart:296:5)
                    #6      BaseTapGestureRecognizer.acceptGesture
                    (package:flutter/src/gestures/tap.dart:267:7)
                    #7      GestureArenaManager.sweep
                    (package:flutter/src/gestures/arena.dart:157:27)
                    #8      GestureBinding.handleEvent
                    (package:flutter/src/gestures/binding.dart:385:20)
                    #9      GestureBinding.dispatchEvent
                    (package:flutter/src/gestures/binding.dart:361:22)
                    #10     RendererBinding.dispatchEvent
                    (package:flutter/src/rendering/binding.dart:278:11)
                    #11     GestureBinding._handlePointerEventImmediately
                    (package:flutter/src/gestures/binding.dart:316:7)
                    #12     GestureBinding.handlePointerEvent
                    (package:flutter/src/gestures/binding.dart:280:5)
                    #13     GestureBinding._flushPointerEventQueue
                    (package:flutter/src/gestures/binding.dart:238:7)
                    #14     GestureBinding._handlePointerDataPacket
                    (package:flutter/src/gestures/binding.dart:221:7)
                    #18     _invoke1 (dart:ui/hooks.dart:180:10)
                    #19     PlatformDispatcher._dispatchPointerDataPacket
                    (dart:ui/platform_dispatcher.dart:276:7)
                    #20     _dispatchPointerDataPacket
(dart:ui/hooks.dart:96:31)
                    (elided 3 frames from dart:async)
                    
                    Handler: "onTap"
                    Recognizer:
                      TapGestureRecognizer#65a4a

```

**Comment by ZainUlMustafa** [comment]
I think it is set to a required property. Will be fixed.

**Comment by ZainUlMustafa** [comment]
Added null checks. https://github.com/ZainUlMustafa/Fluttable/tree/tap-able-cells

Will be released as a main feature in v1.2.3
